great_expectations/datasource/fluent/interfaces.py

```diff
@@ -276,53 +317,36 @@ def delete_batch_config(self, batch_config: BatchConfig) -> None:
         # Let mypy know that self.datasource is a Datasource (it is currently bound to MetaDatasource)
         assert isinstance(self.datasource, Datasource)
 
-        if self.datasource.is_persisted():
-            self.datasource.delete_batch_config(batch_config)
-
         self.batch_configs.remove(batch_config)
-        # If we removed the last of our batch configs, ensure we don't serialize an empty collection
-        # per our serialization conventions.
-        if not self.batch_configs and "batch_configs" in self.__fields_set__:
-            self.__fields_set__.remove("batch_configs")
-
-    def build_batch_request(
-        self,
-        options: Optional[BatchRequestOptions] = None,
-        batch_slice: Optional[BatchSlice] = None,
-    ) -> BatchRequest:
-        """A batch request that can be used to obtain batches for this DataAsset.
-
-        Args:
-            options: A dict that can be used to filter the batch groups returned from the asset.
-                The dict structure depends on the asset type. The available keys for dict can be obtained by
-                calling batch_request_options.
-            batch_slice: A python slice that can be used to limit the sorted batches by index.
-                e.g. `batch_slice = "[-5:]"` will request only the last 5 batches after the options filter is applied.
-
-        Returns:
-            A BatchRequest object that can be used to obtain a batch list from a Datasource by calling the
-            get_batch_list_from_batch_request method.
-        """
-        raise NotImplementedError(
-            """One must implement "build_batch_request" on a DataAsset subclass."""
-        )
-
-    def get_batch_list_from_batch_request(
-        self, batch_request: BatchRequest
-    ) -> List[Batch]:
-        raise NotImplementedError
+        if self.datasource.data_context:
+            try:
+                self.datasource.delete_batch_config(batch_config)
+            except DataContextError:
+                self.batch_configs.append(batch_config)
+                raise
 
-    def _validate_batch_request(self, batch_request: BatchRequest) -> None:
-        """Validates the batch_request has the correct form.
+        self.update_batch_config_field_set()
 
-        Args:
-            batch_request: A batch request object to be validated.
-        """
-        raise NotImplementedError(
-            """One must implement "_validate_batch_request" on a DataAsset subclass."""
-        )
+    def update_batch_config_field_set(self) -> None:
+        """Ensure that we have __fields_set__ set correctly for batch_configs to ensure we serialize IFF needed."""
 
-    # End Abstract Methods
+        has_batch_configs = len(self.batch_configs) > 0
+        if "batch_configs" in self.__fields_set__ and not has_batch_configs:
+            self.__fields_set__.remove("batch_configs")
```

Comment:
Is there a way we can get around having to directly update this attr? `__fields_set__` is an implementation detail within Pydantic right I feel a bit wary touching it if we have an alternative.

Totally get it if that's the path we have to take

Reply:
I really dislike having to touch that as well, but I wasn't able to come up with another solution when I was looking at the same thing with the store approach. And we're already using it in some other places, e.g. `_update__fields_set__on_truthyness`.